### PR TITLE
[RSDK-8331]   Refactor OAK module to use new OAK-FFC-3P config

### PR DESCRIPTION
**Pull request by hexbabe**
[RSDK-8331](https://viam.atlassian.net/browse/RSDK-8331)

tl;dr
The OAK-FFC-3P now works with the new modular config following these changes + the OAK-D still works

### This PR...
- Bumps DepthAI SDK 1.13.1 -> 1.14.0 to get logger changes, deleting the hacky logger handling at the top of `oak.py`
- Refactors loggers to account for their unstable behavior. Saga [here](https://github.com/viamrobotics/viam-python-sdk/pull/690) 
- Adds a TODO for [RSDK-8342](https://viam.atlassian.net/browse/RSDK-8342)
- Changes the name of file `validators.py` to `config.py` for generalization
- Adds `encoders.py` to avoid bloating other files
- Removes individual `Oak` object-scoped fields for config and put all config fields into config objects `OAKDConfig` and `OAKFFC3PConfig`
- Refactors all files to use those config objects instead of manually passing each individual attribute field around
- Adds new `worker.py` state flags for `should_exec` and `configured` to get better visibility of the state of the OakCamera
- Changes default width and height to 720p since @thegreatco  reported that it yields way stabler streams
- Updates integration tests to new model, new defaults, and changed filename

### This PR does not...
- Add a new service model to do ML NN inference stuff
- Update README on how to use OAK-FFC-3P model now that it is available in impl

[RSDK-8331]: https://viam.atlassian.net/browse/RSDK-8331?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### Customer rig emulation OAK-FFC-3P (pickleball setup) on a RPI4
```
{
  "components": [
    {
      "name": "pickleball",
      "namespace": "rdk",
      "type": "camera",
      "model": "viam:luxonis:oak-ffc-3p",
      "attributes": {
        "camera_sensors": [
          {
            "height_px": 720,
            "socket": "cam_c",
            "type": "color",
            "width_px": 1280
          },
          {
            "height_px": 720,
            "socket": "cam_b",
            "type": "color",
            "width_px": 1280
          }
        ]
      }
    }
  ],
  "modules": [
    {
      "type": "local",
      "name": "local-oak-mod",
      "executable_path": "/home/seanyu/test-build-oak-ffc-3p"
    }
  ]
}
```
cam_b
cam_b ✅
cam_c
cam_c ✅
Tested switching primary cameras (the order of the sensors) in the config and it reconfigures properly ✅

Tested `get_images` using a Python client and that worked great. Got two `.show()`able images ✅

### OAK-D integrated model on my mac
```
{
  "components": [
    {
      "name": "poo",
      "namespace": "rdk",
      "type": "camera",
      "model": "viam:camera:oak-d",
      "attributes": {
        "height_px": 720,
        "sensors": [
          "color",
          "depth"
        ],
        "width_px": 1280,
        "frame_rate": 30
      }
    }
  ],
  "modules": [
    {
      "type": "local",
      "name": "oak-mod",
      "executable_path": "/Users/seanyu/Projects/viam-camera-oak/local_run.sh"
    }
  ]
}
```
Tested unplug replug ✅
Tested reconfigure (component name change, attribute change) ✅
Screenshot 2024-07-24 at 1 13 42 PM 
Tested get pcd on app ✅
9a57b558-4c3c-4ad2-85a3-c56e00da29f7
Tested depth stream on app ✅
b0ddaddd-4fcd-45f8-ae81-e84d604c2d49
Tested color stream on app ✅
Tested get_images with my Python client ✅

### Integration test
Built and ran and got `All tests succeeded!` ✅
(on my mac)

[RSDK-8342]: https://viam.atlassian.net/browse/RSDK-8342?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/worker/worker_manager.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
 from threading import Thread
 import time
-from typing import Callable
-from logging import Logger
 
 from src.worker.worker import Worker
 
+# TODO RSDK-8342: re-add the commented logging logic
```
Comment (hexbabe) [inline src/worker/worker_manager.py:6]: Here is the aforementioned TODO for https://viam.atlassian.net/browse/RSDK-8342

**Comment by hexbabe** [comment]
@seanavery I think Rand should review this too. Will add her when she's back

File: src/components/worker/worker.py [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+        return CapturedData(depth_output, timestamp)
+
+    def get_pcd(self) -> CapturedData:
+        if not self.running:
+            raise Exception("Error getting PCD: Worker is not currently running.")
+        try:
+            pc_msg = self.pc_queue_handler.get_queue().get(block=True, timeout=5)
+            if pc_msg.points.nbytes > MAX_GRPC_MESSAGE_BYTE_COUNT:
+                pc_output = self._downsample_pcd(pc_msg.points, pc_msg.points.nbytes)
+            else:
```
Comment (randhid) [inline src/components/worker/worker.py:242]: nice

**Review by randhid — APPROVED** [review]
Honestly LGTM, but I can't give your worker_manager file the attention it needs.

File: src/components/worker/worker.py [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+
+    Args:
+        resolutions (dict)
+    Returns:
+        Union[dai.ColorCameraProperties.SensorResolution, dai.MonoCameraProperties.SensorResolution]
+    """
+
+    def euclidian_distance(width_and_height: Tuple[int, int]) -> float:
+        w1, h1 = width_and_height
+        w2, h2 = width, height
```
[the remark this answers is not in the document]
Comment (hexbabe) [reply src/components/worker/worker.py:73]: We do a check in validation to make sure they're not

File: integration-tests/tests/oak_d_test.go [hunk 4]
```diff
@@ -19,10 +19,10 @@ import (
 )
 
 const (
-	componentName string = "oak-cam"
-	// Default values should mirror those at the top of oak.py and worker.py
-	defaultWidth            int = 640
-	defaultHeight           int = 400
+	componentName string = "oak-d-cam"
```
[the remark this answers is not in the document]
Comment (hexbabe) [reply integration-tests/tests/oak_d_test.go:22]: This change is just me being conscious that we have multiple models floating around now. This integration test tests oak-d only, so I change the component name
